src/Data/Bool/Base.agda

```diff
@@ -72,3 +73,7 @@ T : Bool → Set
 T true  = ⊤
 T false = ⊥
 
+T? : ∀ b → Dec (T b)
```

Comment:
This move represents a non-trivial design decision.  Is `T?` a property or not?  If it's not, then possibly other such moves may be allowed. More likely, it **is** a property, but somehow a very 'basic' one.  So maybe `Data.Bool.Properties.Base` needs to exist for these.

Reply:
I use `blah?` things quite often in programs (i.e., not proofs). E.g., to check if an integer equals `0` before dividing by it. I see that the instances of `=?`  for `Nat` and `Integer` are in their respective properties files. But, I think you could make a case for moving them as they have frequently used computational context, they are decision procedures. Especially if it turns out to be useful to do so.

Reply:
I can definitely warm up to that argument.

Reply:
Hmm so this story is a little more complicated. 

Originally it use to be the case that the decision procedures lived in the `Base` files rather than the `Property` files. The problem was that most decision procedures depend on properties, e.g. the decision procedure for the equality of natural numbers depends on the property that the `suc` constructor is injective, the decision procedure for list permutations depends on the decidedly non-trivial property that if `x :: xs` is a permutation of `ys` then you can extract `x` from `ys` and the result is a permutation of `xs`. All decision procedures were therefore moved to the `Properties` files.

However about a year ago, @laMudri tweaked our definition of `Dec` such that it has both a boolean component and a proof component. @jmchapman when you say that you are using decision procedures in programs rather than proofs, I take it that you're really using the boolean component? I think there's a strong argument that the boolean component of the decision procedure to be in the `Base` file and the proof component to be in the `Properties` file. Indeed we already organise things this way in the numeric classes where we have things like `_≡ᵇ_` and `_<ᵇ_` defined in the `Base` file...

In this case, the proof component of `T?` is trivial so it looks like it can be moved to the `Base` file. However for consistency I'd still think we should have the decision procedure in the `Properties` file. At some point in the next couple of days I'll have a look at how to achieve this...

Reply:
I can re-export the `T?` in the properties file.

The motivation for `T?`'s special status is that `filter` & friends use a decidability
result instead of a boolean function. The only way to recover the `Bool`-based list
functions is to post-compose the filtering boolean predicate with `T?`. Being forced to
pull in all of `Algebra` & `Algebra.Structures` just for that seems a tad heavy.

Reply:
In that case, how about we put `T?` in a new file `Data.Bool.Properties.Base` (ala `Data.Nat.Properties.Core`) rather than in `Data.Bool.Base`?

Reply:
My bad: this was not actually needed!

Reply:
>  @jmchapman when you say that you are using decision procedures in programs rather than proofs, I take it that you're really using the boolean component?

Yes - for the case I was thinking of.